src/toolkit/matl_buy_policy.cc

```diff
@@ -43,18 +45,46 @@ void MatlBuyPolicy::set_manager(Agent* m) {
   }
 }
 
+void MatlBuyPolicy::set_total_inv_tracker(TotalInvTracker* t) {
+  if (t == NULL){
+    std::vector<ResBuf<Material>*> bufs = {buf_};
+    buf_tracker_->Init(bufs, buf_->capacity());
+  }
+  else {
+    buf_tracker_ = t;
+  }
+}
+
+void MatlBuyPolicy::set_inv_policy(std::string inv_policy, double fill, double req_at) {
+  set_req_at(req_at);
+  std::transform(inv_policy.begin(), inv_policy.end(), inv_policy.begin(), ::tolower);
+  if ((inv_policy == "ss")) {
+    set_fill_to(fill);
+  }
+  else if ((inv_policy == "rq") || (inv_policy == "qr")) {
+    set_quantize(fill);
+    // maximum amount that an RQ policy could achieve is req_at + fill
+    set_fill_to(req_at + fill);
+  }
+  else {
+    throw ValueError("Invalid inventory policy");
+  }
+}
+
 void MatlBuyPolicy::set_fill_to(double x) {
-  if (x > 1)
-    x /= buf_->capacity();
-  assert(x > 0 && x <= 1.);
+  assert(x > 0);
+  assert(x <= buf_->capacity());
   fill_to_ = x;
 }
 
-void MatlBuyPolicy::set_req_when_under(double x) {
-  if (x > 1)
-    x /= buf_->capacity();
-  assert(x > 0 && x <= 1.);
-  req_when_under_ = x;
+void MatlBuyPolicy::set_req_at(double x) {
+  assert(x >= 0);
+  req_at_ = x;
```

Comment:
Don't we need to check this against the buffer capacity, too?  Alternatively, we could just set it to the buffer capacity if too large?
```suggestion
  req_at_ = std::min(x, buf_->capacity());
```